frontend/src/lib/UriUtil.ts

```diff
@@ -55,6 +57,45 @@ export function getWindowBaseUriParts(): BaseUriParts {
   return { host, port, basePath }
 }
 
+// NOTE: In the multipage apps world, there is some ambiguity around whether a
+// path like "foo/bar" means
+//   * the page "/" at baseUrlPath "foo/bar", or
+//   * the page "/bar" at baseUrlPath "foo".
+// To resolve this, we just try both possibilities for now, but this leads to
+// the unfortunate consequence of the initial page load when navigating directly
+// to a non-main page of an app being slower than navigating to the main page
+// (as the first attempt at connecting to the server fails the healthcheck).
+//
+// We'll want to improve this situation in the near future, but figuring out
+// the best path forward may be tricky as I wasn't able to come up with an
+// easy solution covering every deployment scenario.
+export function getPossibleBaseUris(): Array<BaseUriParts> {
+  const baseUriParts = getWindowBaseUriParts()
+  const { basePath } = baseUriParts
+
+  if (!basePath) {
+    return [baseUriParts]
+  }
+
+  const parts = basePath.split("/")
+  const possibleBaseUris: Array<BaseUriParts> = []
+
+  while (parts.length > 0) {
+    possibleBaseUris.push({
+      ...baseUriParts,
+      basePath: parts.join("/"),
+    })
+    parts.pop()
+  }
+
+  possibleBaseUris.push({
+    ...baseUriParts,
+    basePath: "",
+  })
+
+  return take(possibleBaseUris, 2)
```

Comment:
From comment: 

```
// NOTE: In the multipage apps world, there is some ambiguity around whether a
// path like "foo/bar" means
//   * the page "/" at baseUrlPath "foo/bar", or
//   * the page "/bar" at baseUrlPath "foo".
```

Do I understand correctly that since 3 level depth is impossible, that's why we take the first two element of  `possibleBaseUris`?

Reply:
We may have a path like `foo/bar/baz`, where `foo/bar` is the `baseUrlPath` and `baz` is the page, but the `pageName` can at most add one part to the path (having a `/` in the `pageName` is illegal), which is why we can return the first two elements of `possibleBaseUris`